### DISCUSSION TITLE
Jobs getting replaced on server start up

### DISCUSSION THREAD

**Discussion by armaniehs** [quartznet/quartznet#1325] [Q&A] [answered]
Hi there,

I noticed that WithMisfireHandlingInstructionIgnoreMisfires() isn't working on version 3.3.3

I used a very simple examples that logs a message every 5 seconds

```
    public static class ConfigureScheduledServicesExtensions
    {
        public static IWebHostBuilder ConfigureScheduledServices(this IWebHostBuilder hostBuilder) =>
            hostBuilder.ConfigureServices((builder, services) => {
                // base configuration from appsettings.json
                services.Configure<QuartzOptions>(builder.Configuration.GetSection("Quartz"));

                services.AddQuartz(q =>
                {
                    q.Properties.Add("quartz.dataSource.riskQuartzDs.connectionString", builder.Configuration["ConnectionStrings:DefaultConnection"]);

                    q.UseTimeZoneConverter();

                    q.UseMicrosoftDependencyInjectionJobFactory();

                    q.ScheduleJob<PingJob>(
                        trigger => trigger
                            .WithIdentity("my", "trigger")
                            .WithCronSchedule("*/5 * * * * ?", cfg => cfg.WithMisfireHandlingInstructionIgnoreMisfires()),
                        job => job
                            .WithIdentity("ping", "pong")
                    );
                });

                // Quartz.Extensions.Hosting hosting
                services.AddQuartzHostedService(options =>
                {
                    // when shutting down we want jobs to complete gracefully
                    options.WaitForJobsToComplete = true;
                });
            });
    }
```

After several hours of investigation I noticed the following log message everytime I restart my server

[07:59:54 INF] [Q.ContainerConfigurationProcessor] Adding 1 jobs, 1 triggers.
[07:59:55 INF] [Q.ContainerConfigurationProcessor] Replacing job: pong.ping

I then added an identity column to the QRTZ_TRIGGERS and QRTZ_CRON_TRIGGERS tables and did confirm that the identity column increases on every server restart

So I commented out the q.ScheduleJob<PingJob> block then restarted the server and then WithMisfireHandlingInstructionIgnoreMisfires works perfectly!

Isn't Quartz suppose to keep the job definition in storage if it hasn't changed since the last server start up or at least honour the NEXT_FIRE_TIME value before job gets replace?

**Comment by lahma** [comment, marked as answer]
Did you notice this part in [documentation](https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/microsoft-di-integration.html#di-aware-job-factories):

```csharp
    // if you are using persistent job store, you might want to alter some options
    services.Configure<QuartzOptions>(options =>
    {
        options.Scheduling.IgnoreDuplicates = true; // default: false
        options.Scheduling.OverWriteExistingData = true; // default: true
    });
```

Seems that you are using persistent job store even though it's not part of your code-based config.

**Reply by armaniehs** [reply]
thanks!

**Reply by armaniehs** [reply]
@lahma I noticed that if I remove a scheduled job from the C# definition it doesn't get removed from the DB on restart. Any hints on the syntax/configuration for it or could you point me out to the documentation, if you can, please?

**Reply by lahma** [reply]
You need to manually remove it calling the `IScheduler` API (`Clear` for all or then individually with `UnscheduleJob`), it would be way too dangerous for Quartz to start removing jobs/triggers it doesn't know of on application start.
